### ISSUE TITLE
LowNodeUtilization does not take into account podAntiAffinity when choosing a pod to evict

### ISSUE THREAD

**Issue by ForbesLindesay** [kubernetes-sigs/descheduler#335]
I have a node that is over-utilized and a node that is under-utilized. There is also a deployment with a pod on both nodes, and an anti-affinity rule that prevents those pods being scheduled on the same node. Unfortunately, each time the descheduler runs, it evicts the pod for this app, which is promptly re-scheduled on the same node, since it cannot be moved to the under-utilized node. There are other pods that could be chosen.

**Comment by lixiang233** [comment]
Yes, descheduler won't check if pods will be properly evicted to other nodes before evicting, so I think it would make sense to add some mechanism to check this and turn to other pods if checks failed.

**Comment by seanmalloy** [comment]
/kind feature

@ForbesLindesay thanks for opening this issue. In general it would be great if the descheduler had the ability to more intelligently select which pods to evict for all strategies. There is a desire to be able to import the kube-scheduler code into the descheduler which would allow leveraging the kube-scheduler's filtering and scoring capabilities. This has been discussed in #184 and #283 

There is some work that needs to be completed before the kube-scheduler can be easily imported. See https://github.com/kubernetes/kubernetes/issues/89930.

**Comment by codablock** [comment]
fyi, the same issue happens with pods that are bound to specific availability zones due to persistent volume claims being only attachable in one specific zone. If such pods are evicted, they will always be scheduled to the same zone, which might mean that it will be the same node if there is only one node per zone.

**Comment by jecnua** [comment]
Still valid.

**Comment by ghostsquad** [comment]
stale-bot is annoying.. 
https://github.com/kubernetes/kubernetes/issues/103151

@lixiang233 - can this issue be marked as a desired feature as to avoid needing to keep it alive (protection from stale-bot)?

**Comment by damemi** [comment]
@ghostsquad I skimmed that thread, but maybe you could point me to what we need to do to mark this as desired? We have other issues that would benefit from that too
